### PR TITLE
Update rules_flex to 0.2.1 and rules_bison to 0.2.2

### PR DESCRIPTION
**Pull request by PshRealtime**
Updates the dependencies flex and bison to latest. Jon Ross-Perkins has stated in the Discord "the patched versions should hopefully go away now that that's in".
There is the concern, that the former URLs are no longer existent.

### CODE REVIEW

**Comment by PshRealtime** [comment]
I am seeing it myself: XXX_register_toolchains should have also `extra_copts = ["-w"]`.

File: WORKSPACE [hunk 1]
```diff
@@ -153,64 +153,47 @@ llvm_zlib_system(name = "zlib")
...
+    sha256 = "10ce41f150ccfbfddc9d2394ee680eb984dc8a3dfea613afd013cfb22ea7445c",
+    urls = ["https://github.com/jmillikin/rules_m4/releases/download/v{0}/rules_m4-v{0}.tar.xz".format(rules_m4_version)],
 )
 
 load("@rules_m4//m4:m4.bzl", "m4_register_toolchains")
 
 # When building M4, disable all compiler warnings as we can't realistically fix
 # them anyways.
-m4_register_toolchains(extra_copts = ["-w"])
+m4_register_toolchains(version = "1.4.18", extra_copts = ["-w"])
```
Comment (chandlerc) [inline WORKSPACE]: Looking at the code, it seems to still support omitting the `version` here and picks the default one?

```suggestion
m4_register_toolchains(extra_copts = ["-w"])
```

File: WORKSPACE [hunk 2]
```diff
@@ -153,64 +153,47 @@ llvm_zlib_system(name = "zlib")
...
+    sha256 = "8929fedc40909d19a4b42548d0785f796c7677dcef8b5d1600b415e5a4a7749f",
+    urls = ["https://github.com/jmillikin/rules_flex/releases/download/v{0}/rules_flex-v{0}.tar.xz".format(rules_flex_version)],
 )
 
 load("@rules_flex//flex:flex.bzl", "flex_register_toolchains")
 
 # When building Flex, disable all compiler warnings as we can't realistically
 # fix them anyways.
-flex_register_toolchains(extra_copts = ["-w"])
+flex_register_toolchains(version = "2.6.4", extra_copts = ["-w"])
```
Comment (chandlerc) [inline WORKSPACE]: Same as above, let's continue to pick up the default:

```suggestion
flex_register_toolchains(extra_copts = ["-w"])
```

File: WORKSPACE [hunk 3]
```diff
@@ -153,64 +153,47 @@ llvm_zlib_system(name = "zlib")
...
+    sha256 = "2279183430e438b2dc77cacd7b1dbb63438971b2411406570f1ddd920b7c9145",
+    urls = ["https://github.com/jmillikin/rules_bison/releases/download/v{0}/rules_bison-v{0}.tar.xz".format(rules_bison_version)],
 )
 
 load("@rules_bison//bison:bison.bzl", "bison_register_toolchains")
 
 # When building Bison, disable all compiler warnings as we can't realistically
 # fix them anyways.
-bison_register_toolchains(extra_copts = ["-w"])
+bison_register_toolchains(version = "3.3.2", extra_copts = ["-w"])
```
Comment (chandlerc) [inline WORKSPACE]:
```suggestion
bison_register_toolchains(extra_copts = ["-w"])
```

**Review by chandlerc — CHANGES_REQUESTED** [review]

**Review by jonmeow — APPROVED** [review]
Thank you for the update! (leaving this in case chandlerc has more comments)

**Comment by PshRealtime** [comment]
There is still some issue with pre-commit. I can't reproduce it on my machine. It appears to me out of the concern with the updating of the dependencies. To trace down the issue will need some more effort.
(This pull request could have also been stalled, because I didn't push the 'Resolve' button...)

@chandlerc Please provide feedback or merge

**Review by chandlerc — APPROVED** [review]
Nope, looks great!

(And sorry, I forgot that requesting changes blocks someone else from approving, my bad!)

**Pull request merged**